scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -3413,7 +3512,16 @@ private void determineCipherSuitesFromConfig() {
 				ciphers.addAll(CipherSuite.getCipherSuitesByKeyExchangeAlgorithm(config.recommendedCipherSuitesOnly,
 						KeyExchangeAlgorithm.PSK));
 			}
-
+			if (config.preselectedCipherSuites != null) {
+				// remove not preselected cipher suites
+				Iterator<CipherSuite> iterator = ciphers.iterator();
+				while (iterator.hasNext()) {
+					CipherSuite cipherSuite = iterator.next();
+					if (!config.preselectedCipherSuites.contains(cipherSuite)) {
+						iterator.remove();
+					}
+				}
```

Comment:
It seems to me that order will be defined by default cipher suite order.
Should we rather use the `preselectedCipherSuites` cipher suite order ?

Reply:
The order only complies to sub-sections in cipher suites. But the order of theses sections is:

- ECDSA
- ECDHE_PSK
- PSK

Anyway, sure we can use the order of the preselected cipher suites.

I adapt it.